### PR TITLE
Movie show

### LINKED ISSUE #8: Movie Details Page
As an authenticated user,
When I visit the movie's detail page,
I should see
- [x] Button to create a viewing party

**Details** This button should take the authenticated user to the new event page

And I should see the following information about the movie:

- [x] Movie Title
- [x] Vote Average of the movie
- [x] Runtime in hours & minutes
- [x] Genere(s) associated to movie
- [x] Summary description
- [x] List the first 10 cast members (characters&actress/actors)
- [x] Count of total reviews
- [x] Each review's author and information

**Details:** This information should come from 3 different endpoints from [The Movie DB API](https://developers.themoviedb.org/3/getting-started/introduction)

### PR DESCRIPTION
**Pull request by jakeheft**
This PR will:
- Set up movie show page with movie attributes
- Add helper method to limit array to 10 results

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -11,4 +11,14 @@ def require_user
     render file: '/public/401' unless current_user
   end
 
+  def hour_min(runtime)
+    hours = runtime / 60
+    rest = runtime % 60
+    "#{hours} hr #{rest} min"
+  end
```
Comment (dcoleman21) [inline app/controllers/application_controller.rb:18]: I wonder if these will eventually live in the service and or the poro we create?

File: app/controllers/movies_controller.rb [hunk 2]
```diff
@@ -1,4 +1,17 @@
 class MoviesController < ApplicationController
+  def show
+    conn = Faraday.new(url: 'https://api.themoviedb.org') do |f|
+      f.params['api_key'] = ENV['MOVIE_API_KEY']
+    end
+    movie_id = params[:id]
+    movie_stats = conn.get("/3/movie/#{movie_id}")
+    @movie_stats = JSON.parse(movie_stats.body, symbolize_names: true)
+    movie_reviews = conn.get("/3/movie/#{movie_id}/reviews")
+    @movie_reviews = JSON.parse(movie_reviews.body, symbolize_names: true)
```
Comment (dcoleman21) [inline app/controllers/movies_controller.rb:10]: These will eventually live in the MovieService that we create (after we do the refactoring lesson)

File: app/views/movies/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<h1> <%=@movie_stats[:title]%> </h1>
+<%= button_to "Create Viewing Party for Movie", '/party/new', method: :get %>
+<p>Vote Average: <%=@movie_stats[:vote_average]%> </p>
```
Comment (dcoleman21) [inline app/views/movies/show.html.erb:3]: These will turn into objects via a poro soon

**Review by dcoleman21 — APPROVED** [review]
Looks great. We will refactor soon

File: app/controllers/movies_controller.rb [hunk 4]
```diff
@@ -1,4 +1,17 @@
...
+    movie_cast = conn.get("/3/movie/#{movie_id}/credits")
+    @movie_cast = JSON.parse(movie_cast.body, symbolize_names: true)
+  end
+
```
Comment (BJSherman80) [inline app/controllers/movies_controller.rb:14]: refactor this method to have less instance methods and maybe not so many api calls

File: app/views/movies/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+<p>Summary: <%=@movie_stats[:overview]%> </p>
+<h2><%= @movie_reviews[:results].count %> Review(s)</h2>
+<% @movie_reviews[:results].each do |review| %>
+  <h3>Author: <%= review[:author] %></h3>
+  <p><%= review[:content] %></p>
+<% end %>
+<h2>Cast</h2>
+<% to_ten(@movie_cast[:cast]).each do |cast| %>
+  <p><%= cast[:name] %> as <%= cast[:character] %></p>
+<% end %>
```
Comment (BJSherman80) [inline app/views/movies/show.html.erb:17]: once we create a poro we will need to refactor this code

File: config/routes.rb [hunk 6]
```diff
@@ -12,6 +12,7 @@
 
   get '/movies', to: 'movies#index'
   get '/movies/search', to: 'movies#search'
+  get '/movies/:id', to: 'movies#show'
```
Comment (BJSherman80) [inline config/routes.rb:15]: we can create a resources for these routes

**Review by BJSherman80 — APPROVED** [review]

**Pull request merged**